acs-integration-tests/src/test/java/com/ge/predix/integration/test/ACSMeteringIT.java

```diff
@@ -101,41 +106,75 @@ public void setup() throws JsonParseException, JsonMappingException, IOException
-            Double beforePolicyEvalMeterCount = getEntitlementUsageByFeatureId(POLICY_EVAL_FEATURE_ID, this.zoneId);
-
-            System.out.println("POLICY UPDATE USAGE BEFORE:" + beforePolicyUpdateMeterCount);
-            System.out.println("POLICY EVAL USAGE BEFORE:" + beforePolicyEvalMeterCount);
-
-            OAuth2RestTemplate acsRestTemplate = this.acsRestTemplateFactory.getACSTemplateWithPolicyScope();
-            String policyFile = "src/test/resources/single-site-based-policy-set.json";
-            testPolicyName = this.policyHelper.setTestPolicy(acsRestTemplate, getZoneHeaders(), this.zoneUrl,
-                    policyFile);
-            ResponseEntity<PolicyEvaluationResult> evalResponse = acsRestTemplate.postForEntity(
-                    this.zoneUrl + PolicyHelper.ACS_POLICY_EVAL_API_PATH,
-                    new HttpEntity<>(policyEvaluationRequest, getZoneHeaders()), PolicyEvaluationResult.class);
-
-            Assert.assertEquals(evalResponse.getStatusCode(), HttpStatus.OK);
-
-            //Nurego server seems to have a lag before the counts are updated
-            Thread.sleep(3000);
 
-            Double afterPolicyUpdateMeterCount = getEntitlementUsageByFeatureId(POLICY_UPDATE_FEATURE_ID, this.zoneId);
-            Double afterPolicyEvalMeterCount = getEntitlementUsageByFeatureId(POLICY_EVAL_FEATURE_ID, this.zoneId);
-
-            System.out.println("POLICY UPDATE USAGE AFTER:" + afterPolicyUpdateMeterCount);
-            System.out.println("POLICY EVAL USAGE AFTER:" + afterPolicyEvalMeterCount);
-
-            // Assert metering counts incremented by 1
-            Assert.assertEquals(afterPolicyUpdateMeterCount - beforePolicyUpdateMeterCount, 1.0);
-            Assert.assertEquals(afterPolicyEvalMeterCount - beforePolicyEvalMeterCount, 1.0);
+        int iteration = 0;
+        try {
+            //Setup a loop to compensate
+            //for slow worker threads on Nurego
+            while(iteration < FINAL_ITERATION) {
+                // Get meter readings before
+                Double beforePolicyUpdateMeterCount = getEntitlementUsageByFeatureId(POLICY_UPDATE_FEATURE_ID, this.zoneId);
+                Double beforePolicyEvalMeterCount = getEntitlementUsageByFeatureId(POLICY_EVAL_FEATURE_ID, this.zoneId);
+
+                LOGGER.info("POLICY UPDATE USAGE BEFORE:" + beforePolicyUpdateMeterCount);
+                LOGGER.info("POLICY EVAL USAGE BEFORE:" + beforePolicyEvalMeterCount);
+
+                OAuth2RestTemplate acsRestTemplate = this.acsRestTemplateFactory.getACSTemplateWithPolicyScope();
+                String policyFile = "src/test/resources/single-site-based-policy-set.json";
+                testPolicyName = this.policyHelper.setTestPolicy(acsRestTemplate, getZoneHeaders(), this.zoneUrl,
+                        policyFile);
+                ResponseEntity<PolicyEvaluationResult> evalResponse = acsRestTemplate.postForEntity(
```

Comment:
Why is this statement inside the loop? It shouldn't be. The intent is to make a single policy evaluation request and loop on the getEntitlementUsageByFeatureId requests to Nurego.